microsoft-edge/webview2/how-to/hostobject.md

```diff
@@ -0,0 +1,262 @@
+
+1. In Visual Studio **Solution Explorer**, open **WebVie2APISample** > **Source Files** > **ScenarioHostObject.cpp**.
+
+1. Go to line 28, where the `ScenarioAddHostObject` class begins.
+
+    ```cpp
+    28    ScenarioAddHostObject::ScenarioAddHostObject(AppWindow* appWindow)
+    29        : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+    30    {
+    31        std::wstring sampleUri = m_appWindow->GetLocalUri(L"ScenarioAddHostObject.html");
+    32
+    33        m_hostObject = Microsoft::WRL::Make<HostObjectSample>(
+    34            [appWindow = m_appWindow](std::function<void(void)> callback)
+    35        {
+    36            appWindow->RunAsync(callback);
+    37        });
+    38
+    39        CHECK_FAILURE(m_webView->add_NavigationStarting(
+    40            Microsoft::WRL::Callback<ICoreWebView2NavigationStartingEventHandler>(
+    41                [this, sampleUri](ICoreWebView2* sender, ICoreWebView2NavigationStartingEventArgs* args) -> HRESULT
+    42        {
+    43            wil::unique_cotaskmem_string navigationTargetUri;
+    44            CHECK_FAILURE(args->get_Uri(&navigationTargetUri));
+    45            std::wstring uriTarget(navigationTargetUri.get());
+    46
+    47            if (AreFileUrisEqual(sampleUri, uriTarget))
+    48            {
+    49                //! [AddHostObjectToScript]
+    50                VARIANT remoteObjectAsVariant = {};
+    51                m_hostObject.query_to<IDispatch>(&remoteObjectAsVariant.pdispVal);
+    52                remoteObjectAsVariant.vt = VT_DISPATCH;
+    53
+    54                // We can call AddHostObjectToScript multiple times in a row without
+    55                // calling RemoveHostObject first. This will replace the previous object
+    56                // with the new object. In our case this is the same object and everything
+    57                // is fine.
+    58                CHECK_FAILURE(
+    59                    m_webView->AddHostObjectToScript(L"sample", &remoteObjectAsVariant));
+    60                remoteObjectAsVariant.pdispVal->Release();
+    61                //! [AddHostObjectToScript]
+    62            }
+    ```
+
+    Lines 31 to 46 show code specific to this sample app where we display HTML. Your app may have a different implementation of this code.  
+
+1. Review line 33, which shows how to instantiate the COM object just defined in the IDL file. This is the object we will use later when we call `AddHostObjectToScript`. This gets us a pointer to the interface in `HostObjectSampleImpl.cpp`.
+
+1. Review line 51, which casts the newly created COM object to an `IDispatch` type and then converts the object to a `VARIANT`. `VARIANT` types allow you to use data structures such as integers and arrays as well as more complex types such as `IDispatch`. 
+
+    For a full list of supported data types, see [VARIANT structure (oaidl.h) - Win32 apps | Microsoft Docs](/windows/win32/api/oaidl/ns-oaidl-variant).
```

Comment:
@jasonstephen15  We should make sure to note that these are types supported by VARIANT itself, but we only support a subset of them, as described in `AddHostObjectToScript` [docs](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2?view=webview2-1.0.1108.44#addhostobjecttoscript).

Reply:
@vchapel pushed content update. Feel free to make wording/format changes as needed. Thanks.